sources.sh

```diff
@@ -219,7 +219,41 @@ jq <<<"$sources" --argjson pins "$externalPinsJson" '
 		# https://unix.stackexchange.com/a/738744/153467
 		reduce .[] as $a ([]; if IN(.[]; $a) then . else . += [$a] end)
 	;
-	reduce .[] as $in ({};
+	def meld($o):
+		# recursive merge of objects like "*", but also append lists (uniquely) instead of replace
+		# https://stackoverflow.com/a/53666584, but with lists unique-ified and less work being done
+		if type == "object" and ($o | type) == "object" then
+			reduce ($o | keys_unsorted[]) as $k (.;
+				.[$k] |= meld($o[$k])
+			)
+		elif type == "array" and ($o | type) == "array" then
+			. + $o
+			| unique_unsorted
+		elif $o == null then
+			.
+		else
+			$o
+		end
+	;
+	(
+		# creating a lookup of .[tag][arch] to a list of sourceIds
+		[
+			.[] as $s
+			| ( $s.arches[] | .tags[], .archTags[] )
+			| {
+				key: .,
+				value: { ($s.arches | keys[]): [$s.sourceId] }
```

Comment:
```suggestion
				value: { ($s.arches | keys[]): [$s.sourceId] },
```

Reply:
Maybe we should also add a comment that this happens pre-arches-merge so looping over arches twice is fine?